NewHorizons/Utility/DebugMenu.cs

```diff
@@ -0,0 +1,321 @@
+            {
+                Logger.Log("potentially updating copy of config at " + filePath);
+
+                if (loadedConfigFiles[filePath].Name == null || AstroObjectLocator.GetAstroObject(loadedConfigFiles[filePath].Name) == null) { Logger.Log("Failed to update copy of config at " + filePath); continue; }      
+
+                var bodyName = loadedConfigFiles[filePath].Name;
+                var astroObjectName = AstroObjectLocator.GetAstroObject(bodyName).name;
+                if (astroObjectName.EndsWith("_Body")) astroObjectName = astroObjectName.Substring(0, astroObjectName.Length-"_Body".Length);
+                var systemName = loadedConfigFiles[filePath].StarSystem;
+                var composedName = systemName + separatorCharacter + astroObjectName;
+
+                planetToConfigPath[composedName] = filePath;
+
+                Logger.Log("made composed name from copy of config file for " + composedName + " " + newDetails.ContainsKey(composedName));
+
+                if (!newDetails.ContainsKey(composedName)) continue;
+
+                
+                if (loadedConfigFiles[filePath].Props == null)
+                {
+                    loadedConfigFiles[filePath].Props = new External.PropModule();
+                }
+                
+                loadedConfigFiles[filePath].Props.Details = newDetails[composedName];
+
+                Logger.Log("successfully updated copy of config file for " + composedName);
+            }
+
+            // find all new planets that do not yet have config paths
+            var planetsThatDoNotHaveConfigFiles = newDetails.Keys.Where(x => !planetToConfigPath.ContainsKey(x)).ToList();
+            foreach (var planetAndSystem in planetsThatDoNotHaveConfigFiles)
+            {
+                Logger.Log("Fabricating new config file for " + planetAndSystem);
+                
+                var filepath = "planets/" + planetAndSystem + ".json";
+                PlanetConfig c = new PlanetConfig(null);
+                c.StarSystem = planetAndSystem.Split(separatorCharacter)[0];
+                c.Name = planetAndSystem.Split(separatorCharacter)[1];
+                c.Props = new PropModule();
+                c.Props.Details = newDetails[planetAndSystem];
+
+                loadedConfigFiles[filepath] = c;
+            }
+        }
+
+        private void InitMenu()
+        {
+            if (_editorMenuStyle != null) return;
+            
+            _dpp = this.GetRequiredComponent<DebugPropPlacer>();
+            _drc = this.GetRequiredComponent<DebugRaycaster>();
+
+
+            Texture2D bgTexture = MakeTexture((int)EditorMenuSize.x, (int)EditorMenuSize.y, Color.black);
+
+            _editorMenuStyle = new GUIStyle
+            {
+                normal =
+                {
+                    background = bgTexture
+                }
+            };
+        }
+
+        private Texture2D MakeTexture(int width, int height, Color color)

```

Comment:
Might want to move this to `Utility/ImageUtilities.cs` so that way we can use it elsewhere if we need to. If you do, name it something like `MakeSolidColorTexture()`.